R/zchunk_L122.ghg_agr_R_S_T_Y.R

```diff
@@ -0,0 +1,194 @@
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L122.EmissShare_R_C_Y_GLU",
+             "L122.ghg_tg_R_agr_C_Y_GLU"))
+  } else if(command == driver.MAKE) {
+
+    all_data <- list(...)[[1]]
+
+    # Load required inputs
+    iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
+    EDGAR_sector <- get_data(all_data, "emissions/EDGAR/EDGAR_sector")
+    L103.ag_Prod_Mt_R_C_Y_GLU <- get_data(all_data, "L103.ag_Prod_Mt_R_C_Y_GLU")
+    L122.LC_bm2_R_HarvCropLand_C_Yh_GLU <- get_data(all_data, "L122.LC_bm2_R_HarvCropLand_C_Yh_GLU")
+    L142.ag_Fert_IO_R_C_Y_GLU <- get_data(all_data, "L142.ag_Fert_IO_R_C_Y_GLU")
+    EDGAR_CH4 <- get_data(all_data, "emissions/EDGAR/EDGAR_CH4") %>%
+      mutate(Non.CO2 = "CH4_AGR")
+    EDGAR_N2O <- get_data(all_data, "emissions/EDGAR/EDGAR_N2O") %>%
+      mutate(Non.CO2 = "N2O_AGR")
+    EDGAR_NH3 <- get_data(all_data, "emissions/EDGAR/EDGAR_NH3") %>%
+      mutate(Non.CO2 = "NH3_AGR")
+    EDGAR_NOx <- get_data(all_data, "emissions/EDGAR/EDGAR_NOx") %>%
+      mutate(Non.CO2 = "NOx_AGR")
+
+    # ===================================================
+    # Compute shares of regional cropland allocation by crop type, and regional production of each crop by each GLU
+    # In the downscaling from (geopolitical) region to crop and GLU, we use land area to go from region to region/crop, and
+    # production to go from region/crop to region/GLU/crop
+
+    # Land area shares (region/crop within region)
+    L122.CropAreaShare_R_C_Y <- L122.LC_bm2_R_HarvCropLand_C_Yh_GLU %>%
+      filter(year %in% HISTORICAL_YEARS) %>%
+      group_by(year, GCAM_region_ID, GCAM_commodity) %>%
+      summarise(value = sum(value)) %>%
+      group_by(year, GCAM_region_ID) %>%
+      # Crop area by region and year
+      mutate(crop_area_total = sum(value)) %>%
+      ungroup() %>%
+      mutate(crop_area_share = value / crop_area_total)
+
+    # Production shares (region/GLU/crop within region/crop)
+    L122.ProdGLUshare_R_C_Y_GLU <- L103.ag_Prod_Mt_R_C_Y_GLU %>%
+      filter(year %in% HISTORICAL_YEARS) %>%
+      group_by(GCAM_region_ID, GCAM_commodity, year) %>%
+      # Production by crop, region, and year
+      mutate(Prod_R_C = sum(value)) %>%
+      ungroup() %>%
+      mutate(prod_share_GLU = value / Prod_R_C) %>%
+      replace_na(list(prod_share_GLU = 0))
+
+    # Emissions shares: product of region/crop shares and region/crop/glu shares
+    L122.EmissShare_R_C_Y_GLU <- L122.ProdGLUshare_R_C_Y_GLU %>%
+      left_join_error_no_match(L122.CropAreaShare_R_C_Y, by = c("GCAM_region_ID", "GCAM_commodity", "year")) %>%
+      transmute(GCAM_region_ID, GCAM_commodity, year, GLU, emiss_share = prod_share_GLU * crop_area_share)
+
+    # Compute EDGAR emissions by region
+    L122.EDGAR <- bind_rows(EDGAR_CH4, EDGAR_N2O, EDGAR_NH3, EDGAR_NOx) %>%
+      # Peat fire and forest fire post burn decay not in EDGAR_sector
+      left_join(EDGAR_sector %>% select(IPCC, sector = agg_sector), by = "IPCC") %>%
```

Comment:
Why not left_join_error_no_match? Do we expect NAs or multiple matches? Clarifying the line 84 comment might help.